shared/blockutil/block_utils.go

```diff
@@ -27,6 +28,28 @@ func SignedBeaconBlockHeaderFromBlock(block *ethpb.SignedBeaconBlock) (*ethpb.Si
 	}, nil
 }
 
+// SignedBeaconBlockHeaderFromBlockInterface function to retrieve signed block header from block.
+func SignedBeaconBlockHeaderFromBlockInterface(block interfaces.SignedBeaconBlock) (*ethpb.SignedBeaconBlockHeader, error) {
+	if block.Block().IsNil() || block.Block().Body().IsNil() {
+		return nil, errors.New("nil block")
+	}
+
+	bodyRoot, err := block.Block().Body().HashTreeRoot()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get body root of block")
+	}
+	return &ethpb.SignedBeaconBlockHeader{
+		Header: &ethpb.BeaconBlockHeader{
+			Slot:          block.Block().Slot(),
+			ProposerIndex: block.Block().ProposerIndex(),
+			ParentRoot:    block.Block().ParentRoot(),
+			StateRoot:     block.Block().StateRoot(),
+			BodyRoot:      bodyRoot[:],
+		},
+		Signature: block.Signature(),
+	}, nil
```

Comment:
```suggestion
func SignedBeaconBlockHeaderFromBlockInterface(sb interfaces.SignedBeaconBlock) (*ethpb.SignedBeaconBlockHeader, error) {
	b := sb.Block()
	if b.IsNil() || b.Body().IsNil() {
		return nil, errors.New("nil block")
	}

	bodyRoot, err := b.Body().HashTreeRoot()
	if err != nil {
		return nil, errors.Wrap(err, "failed to get body root of block")
	}
	h, err := BeaconBlockHeaderFromBlockInterface(b)
	if err != nil {
		return nil, errors.Wrap(err, "failed to get block header of block")
	}
	return &ethpb.SignedBeaconBlockHeader{
		Header: h,
		Signature: sb.Signature(),
	}, nil
```

Suggestion is to reduce the number of times called `Block()` to 1 and call the other helper method to reduce duplicated code.